cg/store/api/findbusinessdata.py

```diff
@@ -164,13 +165,42 @@ def family(self, internal_id: str) -> Family:
         """Fetch a family by internal id from the database."""
         return self.Family.query.filter_by(internal_id=internal_id).first()
 
+    def _get_case_sample_query(self) -> Query:
+        """Return case sample query."""
+        return self.FamilySample.query.join(FamilySample.family, FamilySample.sample)
+
     def family_samples(self, family_id: str) -> List[FamilySample]:
-        """Find the samples of a family."""
-        return (
-            self.FamilySample.query.join(FamilySample.family, FamilySample.sample)
-            .filter(Family.internal_id == family_id)
-            .all()
-        )
+        """Find the samples associated with a case."""
+        return apply_case_sample_filter(
+            function="get_samples_associated_with_case",
+            case_id=family_id,
+            case_samples=self._get_case_sample_query(),
+        ).all()
+
+    def get_sample_cases(self, sample_id: str) -> List[FamilySample]:
+        """Return the cases associated with a sample."""
+        return apply_case_sample_filter(
+            function="get_cases_associated_with_sample",
+            sample_id=sample_id,
+            case_samples=self._get_case_sample_query(),
+        ).all()
+
+    def get_cases_from_sample(self, sample_entry_id: str) -> List[FamilySample]:
+        """Find cases related to a given sample."""
+        return apply_case_sample_filter(
+            function="get_cases_associated_with_sample_by_entry_id",
+            sample_entry_id=sample_entry_id,
+            case_samples=self._get_case_sample_query(),
+        ).all()
+
+    def get_cases_with_samples(self, case_ids: List[str]) -> List[str]:

```

Comment:
Can we rename this function? I feel like `get` suggests a query to the database. Maybe we can use filter, extract or keep?

Reply:
I think there is a query happening, but the return type might be misleading given the name?

Reply:
True, we do query the database 🤔 . Whats bugging me is that we're not asking: Give us all cases with samples (which i feel like the name means). Instead we're asking: which of these given cases has samples.

The question is, what do you name such a function 😅

Reply:
Filter sounds good!